### PR TITLE
`REMOVE_*` commands enhancements

### PR DESCRIPTION
**Pull request by skelly37**
# Summary

* This is a…
  * [x] Bug fix
  * [x] Feature addition
  * [ ] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:
Another batch of useful commands + patch to keep it strictly related to the removal procedure.

# Problem

The commands were left to do later but seemed useful to us

* JIRA ticket (_optional_): None

# Solution

1. Added commands to remove empty clusters and albums & to remove unclustered files
2. Started using Tagger's removal methods to fix unexpected behavior I have encountered during development

# Action

May be squashed imo

### CODE REVIEW

**Comment by rdswift** [comment]
I suspect that the comments on lines 246-247 can be removed?

**Comment by skelly37** [comment]
Ping @phw @zas, fully done imo

File: picard/tagger.py [hunk 1]
```diff
@@ -511,19 +517,28 @@ def handle_command_quit(self, argstring):
         self.quit()
 
     def handle_command_remove(self, argstring):
-        for file in self.iter_all_files():
-            if argstring == file.filename:
-                file.remove()
-                return
+        for file in [x for x in self.iter_all_files() if argstring == x.filename]:
```
Comment (zas) [inline picard/tagger.py:520]: This double loop bothers me, isn't that equivalent to:

```python
for file in self.iter_all_files():
   if file.filename == argstring:
       self.remove([file])
       return
```

**Review by zas — CHANGES_REQUESTED** [review]

File: picard/tagger.py [hunk 2]
```diff
@@ -512,18 +518,28 @@ def handle_command_quit(self, argstring):
...
+                self.remove([file])
                 return
 
     def handle_command_remove_all(self, argstring):
         for file in self.iter_all_files():
-            file.remove()
+            self.remove([file])
+
+    def handle_command_remove_empty(self, argstring):
+        for album in [a for a in self.albums if a.iterfiles()]:
```
Comment (zas) [inline picard/tagger.py:530]: This code looks incorrect to me:
- Album.iterfiles() return an iterator, to evaluate if it has elements a conversion to a list is somehow needed
- the logic seems to be reversed, we want to remove empty albums/clusters

I think the following code is what we need (untested):

```python
for album in self.albums:
   if not any(album.iterfiles()):
       self.remove_album(album)

for cluster in self.clusters:
    if not any(cluster.iterfiles()):
            self.remove_cluster(cluster)
```

**Review by zas — CHANGES_REQUESTED** [review]

**Review by zas — APPROVED** [review]
LGTM

**Review by zas — APPROVED** [review]
LGTM

**Review by phw — APPROVED** [review]

**Pull request merged**